### PR TITLE
Expose type-safe ioctls SNP_PLATFORM_STATUS and SNP_{GET, SET}_EXT_CONFIG

### PR DESCRIPTION
**Pull request by tfanelli-rh**

### CODE REVIEW

**Comment by connorkuehl** [comment]
There is, understandably, a high degree of overlap with the `sev` crate due to some of the abstractions this crate wants to build on top of. I don't think there's anything wrong with that, as it makes the language of both crates consistent with one another.

@npmccallum @haraldh Do you have any objections if the common underlying "scaffolding" for this crate and the `sev` crate are factored out into a `sev-common` crate (or something with more of Enarx's naming flavor, like `sev-ancestor` or `sev-forerunner` or something)?

File: src/util/impl_const_id.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! A simple const generics substitute.
```
Comment (connorkuehl) [inline src/util/impl_const_id.rs:3]: I think min const generics were stabilized at some point.

@npmccallum @haraldh Do you have a preference for how aggressive we should be with how old of a Rust toolchain this crate supports?

File: README.md [hunk 2]
```diff
@@ -5,4 +5,24 @@
 
 # snp
 
+The `snp` crate provides an implementation of [AMD Secure Encrypted
+Virtualization - Secure Nested Paging (SEV-SNP)] APIs.
```
Comment (connorkuehl) [inline README.md:9]: Is this meant to be a link to an SNP page?

Comment (tfanelli-rh) [reply README.md:9]: Yea, i'll add in a link

File: src/util/impl_const_id.rs (hunk 1 above)
Comment (npmccallum) [reply src/util/impl_const_id.rs:3]: We support the use of min const generics (we need it for other things).

I don't think we should bother supporting old Rust. We can even support nightly features (where clearly useful and where there is a path to stabilization). RH is currently rebasing the toolchain regularly, so I don't see much problem with this.

**Comment by npmccallum** [comment]
> There is, understandably, a high degree of overlap with the `sev` crate due to some of the abstractions this crate wants to build on top of. I don't think there's anything wrong with that, as it makes the language of both crates consistent with one another.
> 
> @npmccallum @haraldh Do you have any objections if the common underlying "scaffolding" for this crate and the `sev` crate are factored out into a `sev-common` crate (or something with more of Enarx's naming flavor, like `sev-ancestor` or `sev-forerunner` or something)?

I suspect we probably want an `sev-ioctl` or `iocuddle-sev` package. It looks to me like all the shared stuff is around the ioctls, no?

**Comment by connorkuehl** [comment]
> I suspect we probably want an `sev-ioctl` or `iocuddle-sev` package. It looks to me like all the shared stuff is around the ioctls, no?

Right. I'm thinking at the very least, `sev-iocuddle` would absorb:

* the KVM and SEV ioctl `iocuddle::Group` definitions
* the `Command` and `mut Command` types 
* probably also the `enum Error` since it reflects a value the kernel end up using as a result of the ioctl call
* `Build` and `Version` types

There are probably similarities between the certificate data structures we have in the `sev` crate that might also benefit from being in a common crate, but that's probably where we start to look "off-brand" if the crate is called `sev-iocuddle`.

**Pull request closed without merging**